java/intopt/src/main/java/ru/nsc/interval/thermocompensation/intopt/ThermOpt.java

```diff
@@ -69,7 +69,7 @@ private ResultHolderElement optimizationStep(SetInterval[] box) {
             if (currentSupOfGlobalOptimum < supOfGlobalOptimum) {
                 supOfGlobalOptimum = currentSupOfGlobalOptimum;
                 for (int i = 0; i < result.length; i++) {
-                    result[i] = (int) midBox[i].doubleInf();
```

Comment:
The midBox is a singleton. So midBox[i].doubleInf()=midBox[i].doubleSup().
More important is what we assign to currentSupOfGlobalOptimum.